### PR TITLE
improve documentation and add more examples and fix

### PR DESCRIPTION
**Pull request by 4383**
## Overview

Improve documentation

## Changes

- add more examples on `ifs`, `squote`, `dquote`
- fix YAML sample description following the previous changes introduced by @dj4ngo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,32 +99,73 @@ Output formats are:
...
+```
+
+This is usefull in a shell for loop,
+but your content must, of course, don't contain IFS value:
+```shell
 OIFS="$IFS"
 IFS="|"
-for i in $(niet tests/samples/sample.yaml .project.list-items -f ifs); do
-	echo $i
+for i in $(niet /path/to/your/file.yaml .project.list-items -f ifs); do
```
Comment (dj4ngo) [inline README.md]: Why don't you want to use our samples files ? I used this one to allow to download this file et perform the checks on it. (Maybe we could add the dl URL in this doc for all samples to allow users to test on same files)

File: README.md [hunk 2]
```diff
@@ -158,29 +199,26 @@ Sample example:
...
+    foo: bar
+    list-items:
+        - item1
+        - item2
+        - item3
 ```
 
 Retrieve the project name:
 ```sh
 $ niet tests/samples/sample.yaml project.meta.name
```
Comment (dj4ngo) [inline README.md]: Sometimes we use /path/to/your/file.yaml sometimes tests/samples/sample.yaml.

File: README.md (hunk 1 above)
Comment (4383) [reply README.md]: In accordance to the previous examples in the documentation I prefer to use it along the documentation.

File: README.md (hunk 2 above)
Comment (4383) [reply README.md]: This section was introduced by the fact that we use the sample included in project sources.

**Comment by 4383** [comment]
@dj4ngo please review the update and merge if everything work fine for you!

**Pull request merged**